src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprintAction.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import hudson.model.Run;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import javax.annotation.CheckForNull;
+import jenkins.model.Jenkins;
+import jenkins.model.RunAction2;
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Adds a link, which provides a list of fingerprints with
+ * {@link DockerRunPtrFingerprintFacet}s.
+ *
+ * @author Oleg Nenashev
+ */
+public class DockerFingerprintAction implements RunAction2 {
+
+    private Set<String> imageIDs;
+    transient Run<?, ?> run;
+
+    public DockerFingerprintAction() {
+        this.imageIDs = new HashSet<String>();
+    }
+    
+    public Object readResolve() {
+        if (imageIDs == null) {
+            imageIDs = new HashSet<String>();
```

Comment:
Is this actually needed? You would add such a `readResolve` method (typically `private`, maybe `protected`) when you introduce a new field for which a null/0 default value does not make sense.